packages/grid/x-data-grid-premium/src/hooks/features/export/serializer/excelSerializer.ts

```diff
@@ -0,0 +1,235 @@
+]);
+
+const getFormattedValueOptions = (
+  colDef: GridColDef,
+  valueOptions: ValueOptions[],
+  api: GridApi,
+) => {
+  if (!colDef.valueOptions) {
+    return [];
+  }
+  let valueOptionsFormatted = valueOptions;
+
+  if (colDef.valueFormatter) {
+    valueOptionsFormatted = valueOptionsFormatted.map((option) => {
+      if (typeof option === 'object') {
+        return option;
+      }
+
+      const params: GridValueFormatterParams = { field: colDef.field, api, value: option };
+      return String(colDef.valueFormatter!(params));
+    });
+  }
+  return valueOptionsFormatted.map((option) =>
+    typeof option === 'object' ? option.label : option,
+  );
+};
+
+const serializeRow = (
+  id: GridRowId,
+  columns: GridStateColDef[],
+  api: GridApi,
+  defaultValueOptionsFormulae: { [field: string]: string },
+) => {
+  const row: { [colField: string]: undefined | number | boolean | string | Date } = {};
+  const dataValidation: { [key: string]: Excel.DataValidation } = {};
+
+  const firstCellParams = api.getCellParams(id, columns[0].field);
+  const outlineLevel = firstCellParams.rowNode.depth;
+
+  columns.forEach((column) => {
+    const cellParams = api.getCellParams(id, column.field);
+    switch (cellParams.colDef.type) {
+      case 'singleSelect': {
+        if (typeof cellParams.colDef.valueOptions === 'function') {
+          // If value option depends on the row, set specific options to the cell
+          // This dataValidation is buggy with LibreOffice and does not allow to have coma
+          const valueOptions = cellParams.colDef.valueOptions({ id, row, field: cellParams.field });
+          const formattedValueOptions = getFormattedValueOptions(
+            cellParams.colDef,
+            valueOptions,
+            api,
+          );
+          dataValidation[column.field] = {
+            type: 'list',
+            allowBlank: true,
+            formulae: [
+              `"${formattedValueOptions
+                .map((x) => x.toString().replaceAll(',', 'CHAR(44)'))
+                .join(',')}"`,
+            ],
+          };
+        } else {
+          // If value option is defined for the column, refer to another sheet
+          dataValidation[column.field] = {
+            type: 'list',
+            allowBlank: true,
+            formulae: [defaultValueOptionsFormulae[column.field]],
+          };
+        }
+
+        const formattedValue = api.getCellParams(id, column.field).formattedValue;
+        if (process.env.NODE_ENV !== 'production') {
+          if (String(cellParams.formattedValue) === '[object Object]') {
+            warnInvalidFormattedValue();
+          }
+        }
+        row[column.field] = formattedValue?.label ?? formattedValue;
+        break;
+      }
+      case 'boolean':
+      case 'number':
+      case 'date':
+      case 'dateTime':
+        row[column.field] = api.getCellParams(id, column.field).value;
```

Comment:
Not sure why, but this is how dates look like in Apple Numbers:

<img width="687" alt="Screenshot 2022-04-05 at 18 14 29" src="https://user-images.githubusercontent.com/13808724/161799036-2e32ff30-b62d-4cd7-8681-173a61a3f8a2.png">

This is how it looks like in Google Drive (notice 1-day difference comparing to dates in data grid)
<img width="485" alt="Screenshot 2022-04-05 at 18 15 37" src="https://user-images.githubusercontent.com/13808724/161799303-49e8bb1e-efdf-459c-b4ff-49b3ccc93286.png">